src/Microsoft.Azure.Relay/WebSocketExceptionHelper.cs

```diff
@@ -12,10 +12,17 @@ namespace Microsoft.Azure.Relay
 
     static class WebSocketExceptionHelper
     {
+        public static bool IsRelayContract(Exception exception)
+        {
+            return Fx.IsFatal(exception) ||
```

Comment:
>IsFatal [](start = 22, length = 7)

If it's a Fatal exception then we don't want to wrap it, therefore in a way that's part of our exception contract.

Reply:
I don't know if it's just me, but when a fatal exception comes in, this returns TRUE to it being a relay exception, but we DON'T want it to be a Relay exception...

I know the code makes sense, just takes a bit to wrap my head around

---
In reply to: [279075367](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279075367) [](ancestors = 279075367)

Reply:
It's not "Is it a RelayException" it's "Is it an exception our API should throw?" (aka our Relay API Contract...)

---
In reply to: [279871304](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279871304) [](ancestors = 279871304,279075367)

Reply:
Ok, cool

---
In reply to: [279871945](https://github.com/Azure/azure-relay-dotnet/pull/106#discussion_r279871945) [](ancestors = 279871945,279871304,279075367)